lib/rubocop/cop/lint/ambiguous_block_association.rb

```diff
@@ -30,10 +30,11 @@ class AmbiguousBlockAssociation < Cop
               'call.'
 
         def on_send(node)
-          return if !node.arguments? || node.parenthesized? ||
-                    node.last_argument.lambda? || allowed_method?(node)
+          return unless node.arguments?
```

Comment:
I was surprised to see that this line was still taking non-negligible time, so I investigated a bit and this led to a [bunch](https://github.com/rubocop-hq/rubocop-ast/pull/71) of [fixes](https://github.com/rubocop-hq/rubocop-ast/pull/70) and [a bunch](https://github.com/rubocop-hq/rubocop-ast/pull/68) of [optimizations](https://github.com/rubocop-hq/rubocop-ast/pull/72) in `rubocop-ast` 🎉